crates/astria-conductor/src/executor.rs

```diff
@@ -115,11 +125,54 @@ struct Executor<C> {
     sequencer_hash_to_execution_hash: HashMap<Hash, Vec<u8>>,
 }
 
-impl<C: ExecutionClientV1Alpha1> Executor<C> {
+impl<C: ExecutionClientV1Alpha1 + ExecutionClientV1Alpha2> Executor<C> {
     async fn new(mut execution_rpc_client: C, namespace: Namespace) -> Result<(Self, Sender)> {
         let (cmd_tx, cmd_rx) = mpsc::unbounded_channel();
-        let init_state_response = execution_rpc_client.call_init_state().await?;
-        let execution_state = init_state_response.block_hash;
+
+        let commitment_state = execution_rpc_client.call_get_commitment_state().await?;
+        let soft: Block = commitment_state.soft.unwrap();
+        let firm: Block = commitment_state.firm.unwrap();
+
+        let execution_state = match firm.number.cmp(&soft.number) {
+            Ordering::Equal => {
+                // soft and firm being the same means that the execution chain was just created
+                soft.hash
+            }
+            Ordering::Less => {
+                // get blocks from firm + 1 to soft
+                let identifiers: Vec<BlockIdentifier> = (firm.number + 1..=soft.number)
+                    .map(|block_height| BlockIdentifier {
+                        identifier: Some(Identifier::BlockNumber(block_height)),
+                    })
+                    .collect();
+                let batch_get_blocks_response = execution_rpc_client
+                    .call_batch_get_blocks(identifiers)
+                    .await?;
+                debug!(
+                    "batch_get_blocks_response: {:#?}",
+                    batch_get_blocks_response
+                );
+
+                // TODO - get sequencer hashes for these blocks. best way to do this?
+
+                // TODO - rebuild the hash that tracks sequencer hashes to execution hashes
+
+                // FIXME - what do i actually want to set the execution_state to in this case?
```

Comment:
We might want to update the execution state in general to be a commitment state? If that doesn't make sense, the soft commitment is what we are executing on top of moving forward.